snippets/dart.json

```diff
@@ -0,0 +1,121 @@
+{
+	".source.dart": {
+		"main": {
+			"prefix": "main",
+			"body": [
+				"main(List<String> args) {",
+				"  $1",
+				"}"
+			]
+		},
+		"try": {
+			"prefix": "try",
+			"body": [
+				"try {",
+				"  $2",
+				"} catch (${1:e}) {",
+				"}"
+			]
+		},
+		"if": {
+			"prefix": "if",
+			"body": [
+				"if ($1) {",
+				"  $2",
+				"}"
+			]
+		},
+		"if else": {
+			"prefix": "ife",
+			"body": [
+				"if ($1) {",
+				"  $2",
+				"} else {",
+				"}"
+			]
+		},
+		"switch case": {
+			"prefix": "switch",
+			"body": [
+				"switch ($1) {",
+				"  case $2:",
+				"    $3",
+				"    break;",
+				"  default:",
+				"}"
+			]
+		},
+		"assert": {
+			"prefix": "assert",
+			"body": "assert($1);"
+		},
+		"print": {
+			"prefix": "print",
+			"body": "print($1);"
+		},
+		"for": {
+			"prefix": "for",
+			"body": [
+				"for (var i = 0; i < ${1:count}; i++) {",
+				"  $2",
+				"}"
+			]
+		},
+		"for in": {
+			"prefix": "fori",
+			"body": [
+				"for (var x in ${1:list}) {",
+				"  $2",
+				"}"
+			]
+		},
+		"while": {
+			"prefix": "while",
+			"body": [
+				"while ($1) {",
+				"  $2",
+				"}"
+			]
+		},
+		"do while": {
+			"prefix": "do",
+			"body": [
+				"do {",
+				"  $2",
+				"} while ($1);"
+			]
+		},
+		"void": {
+			"prefix": "void",
+			"body": [
+				"void ${1:name}() {",
+				"  $2",
+				"}"
+			]
+		},
+		"fun": {
+			"prefix": "fun",
+			"body": [
+				"${3:void} ${1:name}(${2:args}) {",
+				"  $4",
+				"}"
+			]
+		},
+		"import": {
+			"prefix": "import",
+			"body": "import \"$1\";"
+		},
+		"class": {
```

Comment:
The `class` one has the same issue that the completion provider takes precedence, which is pretty annoying. I don't think we should remove it though, user will just have to select it (we could filter it out in the completion provider, but that's probably a bit weird!).